src/metabase/driver/sql_jdbc.clj

```diff
@@ -148,14 +157,20 @@
                                :dialect (sql.qp/quote-style driver)))]
     (qp.writeback/execute-write-sql! db-id sql)))
 
-(defmethod driver/truncate! :sql-jdbc
-  [driver db-id table-name]
+(defmethod driver/with-transaction* :sql-jdbc
+  [_ db-id thunk]
+  (jdbc/with-db-transaction [conn (local-conn db-id)]
```

Comment:
wdyt about using `next.jdbc` instead? if yes then we might want to change the arity of this defmulti to add another opts argument for `next.jdbc/with-transcation`

related: https://github.com/metabase/metabase/issues/39549

Reply:
What other options do you anticipate us needing? Since this interface may be used more generally than just JDBC databases I'm not sure we need the power.